protocol/deputy-guardian-module-improvements.md

```diff
@@ -0,0 +1,125 @@
+Council Safe to trigger the Superchain-wide pause function. These powers are given to the
+Foundation Safe through the `DeputyGuardianModule` installed in the Security Council Safe.
+
+## Problem Statement
+
+The existing `DeputyGuardianModule` has a number of problems that we'd like to be able to solve in
+a simple manner that makes as few contract changes as possible. We describe each of these problems
+individually for clarity.
+
+### Pre-Signed Pause Transactions
+
+The Foundation Safe pre-signs certain transactions to reduce response time in case of an emergency.
+Because pre-signed Foundation Safe transactions are dependent on the nonces of that Safe, these
+pre-signed transactions become invalidated any time the Foundation Safe is required to sign
+literally any other transaction. In an attempt to reduce the overhead from this state of affairs,
+a *second* Foundation Safe sometimes referred to as the "Foundation Operations Safe" with the exact
+same configuration as the original Foundation Safe was made to be the Deputy Guardian.
+
+Even with this second Foundation Safe, pre-signed pauses are invalidated on a regular basis
+whenever an upgrade touches the `DeputyGuardianModule`. Gripes with the pre-signed pause system
+could fill a whole role of toilet paper and are not just limited to the issues noted above.
+
+### Unclear Responsibilities
+
+The `DeputyGuardianModule` has an unclear security model and underlying purpose. The original
+intent behind the `DeputyGuardianModule` was to provide a fast way to respond to potential
+incidents in a manner that can impact liveness but not system safety. A lack of strict clarity
+behind this design has meant that the `DeputyGuardianModule` has become a source of security
+implications that are not often easy to reason about.
+
+For instance, the `DeputyGuardianModule` currently allow the Deputy Guardian account to change the
+dispute game type that is respected within the `OptimismPortal` contract from the
+`FaultDisputeGame` to the `PermissionedDisputeGame`. The `PermissionedDisputeGame` has a security
+model that differs from the `FaultDisputeGame` which essentially means that the Deputy Guardian is
+given the ability to change the security model of the system. This is a departure from the original
+design of the `DeputyGuardianModule` which should only impact liveness and should not modify the
+security model of the system.
+
+## Proposed Solution
+
+We propose resolving the above problems by modifying the existing `DeputyGuardianModule` to be more
+restrictive in the actions it can carry out while also being more permissive in who can carry out
+these actions.
+
+The `DeputyGuardianModule` would be permitted to:
+
+1. Trigger the Superchain-wide pause.
+1. Blacklist an individual dispute game.
+
+The `DeputyGuardianModule` would **no longer** be permitted to:
+
+1. Undo the Superchain-wide pause ("unpause").
+1. Change the respected game type.
```

Comment:
Does your opinion here change if we add the `invalidateExistingGames()` function to the `AnchorStateRegistry` and allow the deputy guardian to call that function?

Reply:
I think I could be convinced of allowing the deputy guardian to switch the respected game type, need to think about it a bit more. My primary concern is that if we're making the deputy guardian more permissive then giving it the ability to change the respected game type becomes a bit more dangerous. A malicious/leaked deputy key would be able to swap the respected game type back to the permissionless game which would probably force you to trigger the pause to avoid an endless back-and-forth.

That said I think the likelihood of this happening is relatively low. My short-term proposal would then be that we make the deputy guardian more permissive and give it access to everything except for `unpause`.

Reply:
I was going to come up with a broader proposal but I think the reality is that with Stage 1.4 + interop it's possible that the landscape around security responses looks very different in 6 months, so I'd be happy with a simplified/permissive deputy guardian for now, even if it's not ideal.

Reply:
Yeah I'm also trying to simplify the security model at the same time. I think it's far too many nested permissions and different roles and it's a nightmare to keep track of. I'd really like to be able to keep it down to System Owner + Guardian + Deputy Guardian and nothing else.

> I'd be ok though with the EOA deputy guardian being able to do anything except unpause because it doesn't seem unreasonable to me to pause the superchain if a deputy guardian key was compromised and started being used by an attacker.

This was my thinking too. Compromised key is easy to detect (we can easily monitor for any transactions going through the `DeputyGuardianModule`). It's annoying but it mostly just messes with withdrawals temporarily. We're essentially favoring safety/simplicity over liveness.

Reply:
We could also potentially allow the deputy guardian module to switch the respected game type to permissioned but not allow it to switch back to permissionless.

Reply:
That's actually a pretty good compromise.  When kona is deployed we would typically want to switch to it instead of permissioned if we can so need to think through if that should be allowed by the deputy guardian or if we're ok with requiring the security council for that.  e.g. we could allow the deputy guardian to change the respected game arbitrarily _unless it is already the permissioned game_.